util/reggen/register.py

```diff
@@ -652,3 +653,39 @@ def apply_alias(self, alias_reg: 'Register', where: str) -> None:
 
             # Validate and override attributes.
             field.apply_alias(alias_field, where)
+
+    def scrub_alias(self, where: str) -> None:
+        '''Replaces sensitive fields in register with generic names
+
+        This function can be used to create the generic register descriptions
+        from full alias hjson definitions. It will only work on registers
+        where the alias_target keys are defined, and otherwise throw an error.
+        '''
+        # These attributes are scrubbed
+        assert (self.alias_target)
+        self.name = self.alias_target
+        self.desc = ''
+        self.resval = 0
+        self.tags = []
+        self.alias_target = None
+
+        known_field_names = {}
+        for k, field in enumerate(self.fields):
+            # Infer the aliased field name if it has not been defined.
+            if field.alias_target is None:
+                field.name = 'field{}'.format(k)
+            # Otherwise, use custom alias_target name (make sure it is unique)
+            elif field.alias_target not in known_field_names:
+                field.name = field.alias_target
+                field.alias_target = None
+            else:
+                raise ValueError('Alias field alias_target {} is not unique '
+                                 'in alias register {} in {}.'
+                                 .format(field.alias_target,
+                                         self.alias_target,
+                                         where))
+
+            known_field_names.update({field.name: 1})
```

Comment:
For a slightly better experience if someone calls something `field0`, maybe the right thing is to start by constructing a list of alias targets and checking it doesn't have any duplicates. Then you could check that if `field.alias_target` is not `None` and is of the form `field[0-9]+` then it is exactly equal to `f'field{k}'`, throwing an error if not. Given those two properties, I think you can be certain never to get collisions (or confusingly numbered fields!).

The point is that these checks will report something that's obviously a mistake in the alias definition. The code as it stands might give a very confusing `Alias field alias_target field3 is not unique...` message, leaving the user to stare at their hjson saying "but I only wrote `field3` once!"

Reply:
I've split this into two passes as suggested. PTAL